functions/__z_clean.fish

```diff
@@ -1,9 +1,8 @@
 function __z_clean -d "Clean up .z file to remove paths no longer valid"
-  set -l path (command dirname (status -f))
   set -l tmpfile (mktemp $Z_DATA.XXXXXX)
 
   if test -f $tmpfile
-    command awk -F "|" -f $path/zclean.awk $Z_DATA > $tmpfile
+    command awk -F "|" 'system("test -d \"" $1 "\"") == 0 { print $0 }' $Z_DATA > $tmpfile
```

Comment:
@jethrokuan I think this could be simplified a bit, but I'll leave it for a futuer PR.